### DISCUSSION TITLE
Is it possible to limit the storage (total storage) of a container?

### DISCUSSION THREAD

**Discussion by BasToTheMax** [nestybox/sysbox#645] [Q&A]
^

**Comment by ctalledo** [comment]
Hi @BasToTheMax, sorry for the belated reply. 

Unfortunately I don't believe this is possible with Sysbox currently. 

In general, it's possible to limit container storage in docker via the [`--storage-opt`](https://docs.docker.com/engine/reference/commandline/run/#storage-opt) flag if the underlying filesystem is XFS, but that wouldn't fully work with Sysbox. 

The reason is that Sysbox implicitly allocates volumes under `/var/lib/sysbox` that it then mounts into each container's `/var/lib/docker` and other similar directories. When doing this, Sysbox does not currently limit the size of those volumes, so they are bounded only by the underlying available storage. Ideally Sysbox would honor `--storage-opt` for those volumes (it's something we need to add).

A possible work-around is to mount `/var/lib/sysbox` on a separate storage device (formatted with ext4 preferably) such that you can at least separate the storage used by Sysbox, such so that it won't cause the main filesystem to run out of storage.

Hope that helps.
